### DISCUSSION TITLE
Generate an element without adding it to the graph

### DISCUSSION THREAD

**Discussion by Alexithemia** [cytoscape/cytoscape.js#3015] [Q&A]
Is there any way to generate a full Cytoscape node or edge without it being added to the graph? Like have it exist in a removed state or something?

I have a weird use case for this with the Expand-Collapse extension. I want to create these and add them to the `collapsedChildren` data point of their parent collapsed node, which will then add them to the graph if the parent node is ever expanded.

I have already confirmed I can add the nodes to this data point after they have been removed and everything works as expected, I am just hoping to skip the step of them being added and rendered in the graph first in order to be more optimized.

**Comment by maxkfranz** [comment]
It sounds like this isn't supported by an existing function.  Are you interested in making a PR for this feature?

**Reply by Alexithemia** [reply]
I would be happy to if you think this is possible and would be useful enough to add code for it.

**Reply by maxkfranz** [reply]
It sounds like a good idea.  It ought to be possible, since you can have removed elements -- as you mentioned.  It probably makes sense to start with the constructor by adding support for creating a collection that starts off in the removed state: 

https://github.com/cytoscape/cytoscape.js/blob/56fdab645320b005c9f9f6fce84036641961b850/src/collection/index.js#L25-L121

Then you could create a higher level function that uses your new feature.  Maybe you could reuse the `cy.collection()` function name and allow for passing JSON in like you can with `cy.add()`.  Or maybe there's a better, new name for a function like that.

What do you think?

**Reply by Alexithemia** [reply]
Another small related question, would there be any way to make a removed node a parent of a node? (it would also be a removed node in this case)

It doesn't seem to currently let me make a node that has been collapsed (I.E. removed) by the expand collapse library the parent of a node in the graph using the move functionality.

The use case I want to achieve is:
1. Create a node without adding it to the graph
2. Assign it's parent (that may or may not exist currently in the graph)
3. Add it to the `collapsedChildren` collection data point of the parent node that the extension uses to re-add them to the graph when the parent is expanded.

**Reply by Alexithemia** [reply]
Sure, that all makes sense, I'll start fiddling around with it.

**Reply by maxkfranz** [reply]
Maybe you could allow (2) as a part of (1).  Unit tests would be useful to verify.  (3) sounds like it's up to the extension to manage.

**Reply by Alexithemia** [reply]
Yeah, I have already confirmed (3) works and I will do it externally.

Looking at your element constructor it seems you might already have this functionality. What happens when you don't run restore after creating it? Or at the end of the collection constructor?

https://github.com/cytoscape/cytoscape.js/blob/56fdab645320b005c9f9f6fce84036641961b850/src/collection/element.js#L6-L128
